addon/-private/system/relationships/state/create.js

```diff
@@ -57,23 +57,19 @@ export default class Relationships {
     if (!relationship) {
       let relationshipsByName = get(internalModel.type, 'relationshipsByName');
       let rel = relationshipsByName.get(key);
-      let relationshipPayloadEntry = internalModel.store._relationshipsPayloads.get(internalModel.modelName, internalModel.id, key);
+      let relationshipPayload = internalModel.store._relationshipsPayloads.get(internalModel.modelName, internalModel.id, key);
 
-      if (rel) {
-        relationship = relationships[key] = createRelationshipFor(internalModel, rel, internalModel.store);
-      }
+      if (!rel) { return undefined; }
```

Comment:
should we move this return to before line 60, that way we don't bother getting the relationshipPayload at all?